st2actions/st2actions/scheduler.py

```diff
@@ -56,26 +58,42 @@ def process(self, request):
             LOG.exception('Failed to find liveaction %s in the database.', request.id)
             raise
 
+        # Apply policies defined for the action.
+        for policy_db in Policy.query(resource_ref=liveaction_db.action):
+            policy = policies.get_driver(policy_db.ref,
+                                         policy_db.policy_type,
+                                         **policy_db.parameters)
+            liveaction_db = policy.apply(liveaction_db)
```

Comment:
Actually to make the problem worse is this `for` loop. Since many policies can apply on an action, if we have poor policies defined for an action, we might get something unpredictable.